Mathlib/RingTheory/Polynomial/Basic.lean

```diff
@@ -641,6 +693,23 @@ theorem _root_.Polynomial.coeff_prod_mem_ideal_pow_tsub {ι : Type*} (s : Finset
 
 end CommSemiring
 
+section Semiring
+
+/-- If `R` is a nontrivial ring, the polynomials `R[X]` are not finite as an `R`-module. When `R` is
+a field, this is equivalent to `R[X]` being an infinite-dimensional vector space over `R`.  -/
+theorem polynomial_not_module_finite {R : Type u} [Semiring R] [Nontrivial R] :
```

Comment:
With the current placement this lemma is in the namespace `Ideal`. Is that what you intended? I think I would have moved it up to the other lemmas and called it `not_finite`, so that the full name is `Polynomial.not_finite`. What do you think?

Reply:
Yeah, admittedly I struggled a bit placing this lemma in a meaningful place. But I agree with your suggestion, I will move this around.